moped-editor/src/views/projects/projectView/ProjectComponentEdit.js

```diff
@@ -1,5 +1,5 @@
 import React, { useEffect, useState } from "react";
```

Comment:
a lot of changes here—sorry!

to summarize:
- deprecate `COMPONENT_DETAILS_QUERY`, and move the source for `moped_components` and `moped_subcomonents` up one level in the tree.
- refactor the data munging involved with parsing and packaging up mutation payloads, which are a bit simpler now that we don't have the associative table, `moped_proj_features_components`.
- hookify and move up the tree the functions that index `moped_components` and `moped_subcomponents`

i have to mention that components in general, and specifically the edit state management, was an incredible challenge that Sergio had to solve. these refactors are just polishing his work.